src/org/usfirst/frc/team930/robot/subsystems/SwerveDrive.java

```diff
@@ -0,0 +1,456 @@
+			// System.out.println("D2 " + D2);
+
+			oldTopRightAngle = topRightAngle;
+			oldTopLeftAngle = topLeftAngle;
+			oldBottomRightAngle = bottomRightAngle;
+			oldBottomLeftAngle = bottomLeftAngle;
+
+			topRightSpeed = Math.sqrt(B2 + C2);
+			topLeftSpeed = Math.sqrt(B2 + D2);
+			bottomLeftSpeed = Math.sqrt(A2 + D2);
+			bottomRightSpeed = Math.sqrt(A2 + C2);
+
+			topRightAngle = (int) (Math.atan2(B, C) * RAD_TO_DEG);
+			topLeftAngle = (int) (Math.atan2(B, D) * RAD_TO_DEG);
+			bottomLeftAngle = (int) (Math.atan2(A, D) * RAD_TO_DEG);
+			bottomRightAngle = (int) (Math.atan2(A, C) * RAD_TO_DEG);
+
+		}
+
+		int modTopRight = (int) ((oldTopRightAngle - topRightAngle) / 360);
+		int modTopLeft = (int) ((oldTopLeftAngle - topLeftAngle) / 360);
+		int modBottomRight = (int) ((oldBottomRightAngle - bottomRightAngle) / 360);
+		int modBottomLeft = (int) ((oldBottomLeftAngle - bottomLeftAngle) / 360);
+
+		double diffTopRight = (oldTopRightAngle - topRightAngle) % 360;
+		double diffTopLeft = (oldTopLeftAngle - topLeftAngle) % 360;
+		double diffBottomRight = (oldBottomRightAngle - bottomRightAngle) % 360;
+		double diffBottomLeft = (oldBottomLeftAngle - bottomLeftAngle) % 360;
+
+		 /*
+		 * So this does run in O(1) since the while loop will only run once
+		 * or twice due to the += 360 * (int) lines of code to bring it within
+		 * 360 degrees. For all intents and purposes it won't run that long
+		 *
+		 * WARNING: CURRENTLY SLIGHTLY BROKEN. WORKING ON FIX
+		 */
+		int dir;
+		// tr
+		System.out.println("topRightAngle b4 it is set " + topRightAngle);
+		topRightAngle += 360 * (int) ((oldTopRightAngle - topRightAngle) / 360);
+		System.out.println("old tr angle " + oldTopRightAngle);
+		System.out.println("old tr speed " + topRightSpeed);
+
+		
+		if (oldTopRightAngle > topRightAngle) {
+			dir = 1;
+		} else {
+			dir = -1;
+		}
+		while (Math.abs(oldTopRightAngle - topRightAngle) > 90) {
+			topRightAngle += dir * 180;
+			topRightSpeed *= -1;
+
+			System.out.println("tr angle " + topRightAngle);
+			System.out.println("tr speed " + topRightSpeed);
+
+		}
+		 // tl
+		 topLeftAngle += 360 * (int) (oldTopLeftAngle - topLeftAngle);
+		 if (oldTopLeftAngle > topLeftAngle)
+		 dir = 1;
+		 else
+		 dir = -1;
+		 while (Math.abs(oldTopLeftAngle - topLeftAngle) > 90) {
+		 topLeftAngle += dir * 180;
+		 topLeftSpeed *= -1;
+		 }
+		 // bl
+		 bottomLeftAngle += 360 * (int) (oldBottomLeftAngle - bottomLeftAngle);
+		 if (oldBottomLeftAngle > bottomLeftAngle)
+		 dir = 1;
+		 else
+		 dir = -1;
+		 while (Math.abs(oldBottomLeftAngle - bottomLeftAngle) > 90) {
+		 bottomLeftAngle += dir * 180;
+		 bottomLeftSpeed *= -1;
+		 }
+		 // br
+		 bottomRightAngle += 360* (int) (oldBottomRightAngle - bottomRightAngle);
+		 if (oldBottomRightAngle > bottomRightAngle)
+		 dir = 1;
+		 else
+		 dir = -1;
+		 while (Math.abs(oldBottomRightAngle - bottomRightAngle) > 90) {
+		 bottomRightAngle += dir * 180;
+		 bottomRightSpeed *= -1;
+		 }
+
```

Comment:
This commented area does not look like it is needed any longer. Can be removed unless otherwise stated.